test/sql/setops/union_limit.test

```diff
@@ -0,0 +1,194 @@
+# name: test/sql/setops/union_limit.test
+# description: Test union limit
+# group: [setops]
+
+statement ok
+PRAGMA enable_verification
+
+query I
+SELECT 1 UNION ALL SELECT * FROM range(2, 100) UNION ALL SELECT 999 LIMIT 5;
+----
+1
+2
+3
+4
+5
+
+query I
+SELECT 'select count(case'a union all select 'when a='||range||' then 1' from range(100) LIMIT 5;
+----
+select count(case
+when a=0 then 1
+when a=1 then 1
+when a=2 then 1
+when a=3 then 1
+
+query I
+SELECT STRING_AGG(a, ' ') FROM (SELECT 'select count(case'a union all select 'when a='||range||' then 1' from range(2) union all select 'end) from t') tbl;
+----
+select count(case when a=0 then 1 when a=1 then 1 end) from t
+
+query I
+SELECT 'select count(case'a union all select 'when a='||range||' then 1' from range(2) union all select 'end) from t' LIMIT 5;
+----
+select count(case
+when a=0 then 1
+when a=1 then 1
+end) from t
+
+query I
+SELECT 'select count(case'a union all select 'when a='||range||' then 1' from range(100) union all select 'end) from t' LIMIT 5;
+----
+select count(case
+when a=0 then 1
+when a=1 then 1
+when a=2 then 1
+when a=3 then 1
+
+
+query I
```

Comment:
It might. be nice to have a short comment on what each of these queries is testing. It looks to me like the problem was that different levels of nesting were getting mixed up?